### PR TITLE
feat(cache) allow configuring neg_ttl for caching db entities

### PR DESCRIPTION
**Pull request by ealogar**
### Summary 

The cache allows to configure ttl and neg_ttl but kong impose that the values are equal. With this change, any can set both values when this is needed.
For example any can configure a neg_ttl to be lower than ttl that is more useful in most of the cases.
When the datase is down and a nil value is cached, this stay until normal ttl expires and this may lead to not recover quickly when database is up.
Setting this neg_ttl to a lower value than ttl can make this recover quickly.
If the value is not set, cache_ttl will be used as negative ttl (as it's done now)

### Full changelog

* add a new 'db_cache_neg_ttl' config option

### CODE REVIEW

**Comment by ealogar** [comment]
cc @hishamhm , I opened a new pr with your requested changes

**Comment by hishamhm** [comment]
@ealogar Thank you for addressing the review! We are going to run some local tests against this branch to make sure the behavior on database unavailability is the expected one.

**Comment by ealogar** [comment]
@hishamhm Good, i already performed the tests

File: kong.conf.default [hunk 1]
```diff
@@ -874,6 +874,13 @@
                                  # If set to 0 (default), such cached entities
                                  # or misses never expire.
 
+
+#db_cache_neg_ttl = 0            # Time-to-live (in seconds) of database misses
+                                 # where no entity is returned and nil is cached.
+                                 #
+                                 # If not set, the value of db_cache_ttl will be
+                                 # used.
```
Comment (thibaultcha) [inline kong.conf.default]: Let's update the description to be more accurate, more detailed, and better aligned with its `db_cache_ttl` counterpart:

```
Time-to-live (in seconds) of a datastore miss (no entity).

If not specified (default), the `db_cache_ttl` value will be used instead.

If set to 0, datastore misses will never expire.
```

File: kong.conf.default [hunk 2]
```diff
@@ -874,6 +874,13 @@
                                  # If set to 0 (default), such cached entities
                                  # or misses never expire.
 
+
```
[the remark this answers is not in the document]
Comment (thibaultcha) [reply kong.conf.default]: Additionally, the second line of the above `db_cache_ttl` property's description should now mention this new property.

**Comment by ealogar** [comment]
@thibaultcha your requested changes are done.

**Comment by bungle** [comment]
@ealogar can you rebase this to latest `next` so that we get a green `travis`?

**Comment by ealogar** [comment]
@bungle green travis already ...

**Comment by kikito** [comment]
Thanks a lot for this, it has been merged into `next`!

**Pull request merged**